src/components/PopupWithForm.js

```diff
@@ -32,12 +34,15 @@ export default class PopupWithForm extends Popup {
     super.removeEventListeners();
   }
 
-  _handleSubmitForm = () => {
+  _handleSubmitForm = (evt) => {
+    this._submitBtn.textContent = 'Saving...';
     this._callback(this._getInputValues());
-    this.close();
+    evt.preventDefault();
   };
 
   reset() {
+    this.close();
```

Comment:
Needs correction: this method is responsible for resetting the form. You should close the popup in index.js after successful api response in `then` block